packages/netsuite-adapter/src/filters/convert_lists_to_maps.ts

```diff
@@ -0,0 +1,49 @@
+/*
+*                      Copyright 2021 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import {
+  isInstanceElement, isObjectType,
+} from '@salto-io/adapter-api'
+import { collections } from '@salto-io/lowerdash'
+import { convertFieldsTypesFromListToMap, convertInstanceListsToMaps } from '../mapped_lists/mapped_lists'
+import { FilterWith } from '../filter'
+
+const { awu } = collections.asynciterable
+
+const filterCreator = (): FilterWith<'onFetch'> => ({
+  /**
+   * Upon fetch, convert values of list type to maps
+   *
+   * @param elements the already fetched elements
+   */
+  onFetch: async elements => {
+    await awu(elements)
+      .filter(isObjectType)
+      .forEach(
+        async type => convertFieldsTypesFromListToMap(type)
+      )
```

Comment:
nit: I **think** that this one is fine since you don't really have to state that it's async, as it doesn't use `await` inside.
```suggestion
      .forEach(convertFieldsTypesFromListToMap)
```